spec/client/Rollup-Integration-guide.md

```diff
@@ -0,0 +1,201 @@
+
+Rollups on the other hand do have a fallback layer in the data availability and settlement layers. For example, the settlement layer can verify a block is invalid and simply remove it thus enforcing that blocks can keep proceeding with valid states as the settlement layer can continue removing invalid blocks from the chain history. Similarly, it's possible that the settlement layer has a mechanism to switch the sequencer if a block is proven invalid.
+
+Thus, `updateStateOnMisbehaviour` can be less strict for rollups and simply remove the fraudulent consensus state and wait for the resolution as specified by the rollup's consensus rules.
+
+```typescript
+function updateStateOnMisbehaviour(clientMessage: ClientMessage) {
+    // unmarshalling logic ommitted
+    misbehaviour = clientMessage.misbehaviour
+    misbehavingHeight = getHeight(misbehaviour)
+
+    // delete the fraudulent consensus state
+    deleteConsensusState(clientMessage.clientId, misbehavingHeight)
+
+    // its possible for the rollup client to do additional logic here
+    // e.g. verify the next sequencer chosen from settlement layer
+    // however this is highly specific to rollup architectures
+    // and is not necessary for all rollup architectures
+    // so it will not be modelled here.
+}
+```
+
+### VerifyMembership Methods
+
+The parts of the client that rely on the data availability are encapsulated in verifying new rollup blocks. Thus, once they are already added to the client, they can be used for proof verification without reference to the underlying data availability layer. For optimistic rollups, the consensus state must exist in the client for the full fraud period before it can be used for proof verification.
+
+Since the rollup client is dependent on underlying clients: data availability client and settlement client, these must also not be frozen by misbehaviour in order for proof verification to proceed.
+
+```typescript
+function verifyMembership(
+    clientState: ClientState,
+    height: Height,
+    delayPeriodTime: uint64, // disabled
+    delayPeriodBlocks: uint64, // disabled
+    proof: CommitmentProof,
+    path: CommitmentPath,
+    value: bytes
+): error {
+    // check conditional clients are still valid
+    rollupClient = getClient(clientState.DALayer)
+    settlementClient = getClient(clientState.settlementLayer) // may not exist for all rollups
+
+    assert(isActive(clientState))
+    assert(isActive(rollupClient))
+    assert(isActive(settlementClient))
+
+    consensusState = getConsensusState(clientState.clientId, height)
+    submitTime = getSubmitTime(consensusState)
+
+    // must ensure fraud proof period has passed
+    assert(submitTime + clientState.fraudPeriod > currentTime())
```

Comment:
Is `fraudPeriod` then something we should store in the `ClientState` of the Rollkit client?

Reply:
Yes, we could theoretically put it in the connection and encode as the delay time. But I think that is more confusing.

We should discuss with rollup teams as to what makes more sense